rclcpp/include/rclcpp/strategies/static_memory_strategy.hpp

```diff
@@ -28,6 +28,12 @@ namespace memory_strategies
 namespace static_memory_strategy
 {
 
+template<size_t max_subscriptions_ = 10,
+size_t max_services_ = 10,
+size_t max_clients_ = 10,
+size_t max_executables_ = 1,
+size_t max_guard_conditions_ = 2,
+size_t pool_size_ = 1024>
```

Comment:
A constant like this should be upper case.

But I don't see why this need to be a template argument. A constructor argument seems to be sufficient.

The constructor could take a struct with these limits and there can be a default value for it with the above limits.